scripts/traverse.js

```diff
@@ -15,124 +15,169 @@ import bcd from '../index.js';
-        ]
-      : folder.split(',');
-  const values = Array.isArray(value) ? value : value.toString().split(',');
 
-  for (const folder in folders)
+  for (const folder in folders) {
     features.push(
       ...traverseFeatures(
         bcd[folders[folder]],
-        depth,
+        browsers,
         values,
-        browser,
-        `${folders[folder]}.`,
+        depth,
+        folders[folder] + '.',
       ),
     );
+  }
 
-  console.log(features.join('\n'));
-  console.log(features.length);
+  return features;
 };
 
 if (esMain(import.meta)) {
   const { argv } = yargs(hideBin(process.argv)).command(
-    '$0 <browser> [folder] [value]',
-    'Test for specified values in any specified browser',
+    '$0 [folder...]',
+    'Print feature names in the folder (and optionally filter features to specific browser or version values)',
     (yargs) => {
       yargs
-        .positional('browser', {
-          describe: 'The browser to test for',
-          type: 'string',
-        })
         .positional('folder', {
-          describe: 'The folder(s) to test (set to "all" for all folders)',
+          describe: 'The folder(s) to traverse',
           type: 'array',
-          default: 'all',
+          default: Object.keys(bcd).filter((k) => k !== 'browsers'),
         })
-        .positional('value', {
-          describe: 'The value(s) to test against',
+        .option('browser', {
+          alias: 'b',
+          describe: 'Filter by a browser. May repeat.',
           type: 'array',
-          default: ['null', 'true'],
+          nargs: 1,
+          default: Object.keys(bcd.browsers),
+        })
+        .option('filter', {
+          alias: 'f',
+          describe: 'Filter by version value. May repeat.',
+          type: 'array',
+          nargs: 1,
+          default: [],
+        })
+        .option('non-real', {
+          describe:
+            'Filter to features with non-real values. Alias for "-f true -f null"',
+          type: 'boolean',
+          nargs: 0,
         })
         .option('depth', {
           alias: 'd',
           describe:
             'Depth of features to traverse (ex. "2" will capture "api.CSSStyleSheet.insertRule" but not "api.CSSStyleSheet.insertRule.optional_index")',
           type: 'number',
-          default: 100,
-        });
+          nargs: 1,
+          default: 10,
+        })
+        .example(
+          'npm run traverse -- --browser=safari --non-real',
+          'Find all features containing non-real Safari entries',
+        )
+        .example(
+          'npm run traverse -- -b webview_android -f true',
+          'Find all features marked as true for WebVeiw',
+        )
+        .example(
+          'npm run traverse -- -b firefox -f 10',
+          'Find all features marked as supported since Firefox 10',
+        );
     },
   );
 
-  main(argv.folder, argv.value, argv.browser, argv.depth);
+  const features = main(argv.folder, argv.value, argv.browser, argv.depth);
```

Comment:
The order of `argv.value` and `argv.browser` here are flipped, and that's one cause of https://github.com/mdn/browser-compat-data/issues/16505. But even fixing that none of the filtering seems to work.